pkg/render/policyrecommendation_test.go

```diff
@@ -347,11 +353,30 @@ var _ = Describe("Policy recommendation rendering tests", func() {
 			for i, expectedRes := range tenantBExpectedResources {
 				rtest.ExpectResourceTypeAndObjectMetadata(tenantBResources[i], expectedRes.name, expectedRes.ns, expectedRes.group, expectedRes.version, expectedRes.kind)
 			}
-			d = rtest.GetResource(tenantBResources, "tigera-policy-recommendation", tenantBNamespace, appsv1.GroupName, "v1", "Deployment").(*appsv1.Deployment)
-			envs = d.Spec.Template.Spec.Containers[0].Env
-			Expect(envs).To(ContainElement(corev1.EnvVar{Name: "TENANT_NAMESPACE", Value: tenantBNamespace}))
-			Expect(envs).To(ContainElement(corev1.EnvVar{Name: "TENANT_ID", Value: "tenant-b-id"}))
-
+			verifyTenantResources(tenantBResources, tenantBNamespace, "tenant-b-id")
 		})
 	})
 })
+
+func verifyTenantResources(createdResources []client.Object, namespace string, tenantID string) {
```

Comment:
This is somewhat personal preference, but I would really like if we avoid these types of catch-all helper functions in tests for a few reasons:

- It makes test output and line numbers convoluted and harder to follow (ExpectWithOffset helps a bit, but not much)
- It assumes every test wants to test the same thing, which isn't the case. Especially in the future we may alter expectations which will require making this function even more complicated, or would require that we write tests that don't use this function anyway.
- It makes it harder to read through a test and understand what the test is actually trying to verify.

I would _much_ rather we write targeted tests. There's no need for _every_ test to assert that the clusterrole / binding is done correctly - let's just have one test that asserts that's correct in a multi-tenant environment. Then it becomes really clear what's going on when it fails. With this model, if we introduce a bug into the RBAC resources, every multi-tenant UT will fail and it will be less obvious what the problem is.

Reply:
Np. I can create a new test to check RBAC. Maybe we should rename this test from `should render expected components inside expected namespace for each policyrecommendation instance` to something more specific then ? Seems like we are checking for components and some environment variables.

Reply:
I think it's good / fine to have one test that verifies the high level stuff - the correct resources were rendered - but then we should have fine-grained tests that verify specific inputs map to specific outputs.